ui_testing/testcases/basic_tests/test06_orders.py

```diff
@@ -970,109 +974,32 @@ def test023_create_existing_order_with_test_units_and_change_article(self):
-                                                                                                                     len(
-                                                                                                                         rows_count)))
-        self.assertEqual(len(rows_count), 3)
-
-        # self.base_selenium.refresh()
-        order_url = self.base_selenium.get_url()
-        self.base_selenium.get(
-            url=order_url, sleep=self.base_selenium.TIME_MEDIUM)
-
-        self.order_page.archive_suborder(index=suborder_index_to_archive, check_pop_up=True)
-
-        self.base_selenium.get(
-            url=order_url, sleep=self.base_selenium.TIME_MEDIUM)
-        self.order_page.get_suborder_table()
-        rows_count = self.base_selenium.get_table_rows(element='order:suborder_table')
-
-        self.base_selenium.LOGGER.info('+ Number of suborders = {}'.format(len(rows_count)))
-        self.base_selenium.LOGGER.info(
-            '+ Assert  before archiving suborders count {}, after archiving suborders count {}'.format(3,
-                                                                                                       len(rows_count)))
-        self.assertEqual(len(rows_count), 2)
-
-        self.order_page.get_orders_page()
+        LIMS-6518
+        """
+        self.base_selenium.LOGGER.info('select random record')
+        orders, payload = self.orders_api.get_all_orders(limit=20)
+        order_data = random.choice(orders['orders'])
+
+        order_no = order_data['orderNo']
+        self.order_page.apply_filter_scenario(filter_element='orders:filter_order_no', filter_text=order_no, field_type='text')
+        suborders_data = self.order_page.get_child_table_data(index=0)
+        self.order_page.archive_table_suborder(index=0)
+        self.base_selenium.LOGGER.info('make sure that suborder is archived successfully')
+        if len(suborders_data) > 1:
+            suborder_data_after_archive = self.order_page.get_table_data()
+            self.assertNotEqual(suborder_data_after_archive[0], suborders_data[0])
+        else:
+            table_records_count = len(self.order_page.result_table()) -1
+            self.assertEqual(table_records_count, 0)
 
-        # going to archived orders page to make sure that the order is archived
         self.order_page.get_archived_items()
-
-        self.order_page.open_filter_menu()
-        self.orders_page.filter_by_analysis_number(filter_text=analysis_no)
-
-        rows_count = self.order_page.result_table()
-        self.base_selenium.LOGGER.info(
-            '+ Assert count archived orders with a specific analysis number: {}'.format(len(rows_count)))
-
-        self.assertEqual(len(rows_count) - 1, 1)
+        self.order_page.search(order_no)
+        results = self.order_page.result_table()[0].text
+        self.assertIn(order_no.replace("'", ""), results.replace("'", ""))
```

Comment:
the same here if order has multiple suborders and one of them was archived before this assertion will raise true even if suborder not archived
you need to search by analysis no

Reply:
@ghada-37 I will assert with the analysis number in the two test case and abut the previous comment ( if order has another suborder, this assertion will be true even if suborder not restored
so, you need to make assertion by analysis no not order no ) it the same comment you mean right

Reply:
it's the same comment yeah., assertion or search whatever the method you select it should depend on anaysis no

Reply:
Done